test/compute-engine/canonical-form.test.ts

```diff
@@ -286,11 +321,18 @@ describe('CANONICAL FORMS', () => {
         canonForms = NaN
         canonical  = NaN
       `);
+      //↓Only simplifies because 'x' substitutes with its value during partial-canonicalization;
+      //prior to app. of CanonicalForms.
       expect(checkPower('16^x')).toMatchInlineSnapshot(`
         box        = ["Power", 16, "x"]
         canonForms = 1
         canonical  = 1
       `);
+      //!@note: this should be ok to apply, despite 'Pi' being a constant which does *not* have a
+      //!'holdUntil' value set to 'never', because... this is a library/engine-level defined
+      //!constant in which its type/value is always known: even in non-canonical expressions.
+      //I.e., if this were a user-defined constant, this would not be OK, since its value would have
+      //to be referenced.
```

Comment:
This case relates to the second bullet printed in the PR opening.
Seems reasonable to me that this proceeds, as an exception, given that it is an engine/inbuilt *library* definition?

Reply:
I agree.